controllers/resources.go

```diff
@@ -0,0 +1,230 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	crossplanecore "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	compute "github.com/crossplaneio/crossplane/apis/compute/v1alpha1"
+	database "github.com/crossplaneio/crossplane/apis/database/v1alpha1"
+	workload "github.com/crossplaneio/crossplane/apis/workload/v1alpha1"
+
+	"github.com/crossplaneio/sample-stack-wordpress/api/v1alpha1"
+)
+
+var (
+	// Used by the resources in managed Kubernetes cluster where Wordpress
+	// actually runs.
+	labelsInRemote = map[string]string{
+		"app": "wordpress",
+	}
+	// Used by the resources in the cluster where WordpressInstance custom
+	// resource resides.
+	labelsInLocal = map[string]string{
+		"stack": "sample-stack-wordpress",
+	}
+)
+
+func GetKubernetesCluster(wp v1alpha1.WordpressInstance) compute.KubernetesCluster {
+	claimName := fmt.Sprintf("wordpress-cluster-%s-%s", wp.GetNamespace(), wp.GetName())
+	return compute.KubernetesCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            claimName,
+			Namespace:       wp.GetNamespace(),
+			Labels:          labelsInLocal,
+			OwnerReferences: []metav1.OwnerReference{GetOwnerReference(wp)},
+		},
+		Spec: compute.KubernetesClusterSpec{
+			ResourceClaimSpec: crossplanecore.ResourceClaimSpec{
+				WriteConnectionSecretToReference: &crossplanecore.LocalSecretReference{
+					Name: claimName,
+				},
+			},
+		},
+	}
+}
+
+func GetMySQLInstance(wp v1alpha1.WordpressInstance) database.MySQLInstance {
+	claimName := fmt.Sprintf("wordpress-sql-%s-%s", wp.GetNamespace(), wp.GetName())
+	return database.MySQLInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            claimName,
+			Namespace:       wp.GetNamespace(),
+			Labels:          labelsInLocal,
+			OwnerReferences: []metav1.OwnerReference{GetOwnerReference(wp)},
+		},
+		Spec: database.MySQLInstanceSpec{
+			ResourceClaimSpec: crossplanecore.ResourceClaimSpec{
+				WriteConnectionSecretToReference: &crossplanecore.LocalSecretReference{
+					Name: claimName,
+				},
+			},
+			EngineVersion: "5.7",
+		},
+	}
+}
+
+func GetKubernetesApplication(wp v1alpha1.WordpressInstance, dbClaimName string) workload.KubernetesApplication {
+	namespaceResource := ConvertToUnstructured(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "wordpress",
+			Labels: labelsInRemote,
+		},
+	})
+	localDeploymentResourceName := fmt.Sprintf("wordpress-demo-deployment-%s-%s", wp.GetNamespace(), wp.GetName())
```

Comment:
Now that we've moved beyond a heredoc with a template, I am not a huge fan of the hard-coded string prefixes, since they're a bit harder to work with if we ever want to change them . . . what are your thoughts on pulling them up into constants somewhere so they're easier to change?